cpp/src/arrow/engine/substrait/relation_internal.cc

```diff
@@ -94,6 +94,20 @@ Status CheckRelCommon(const RelMessage& rel) {
   return Status::OK();
 }
 
+// Other helper functions
+Status DiscoverFilesFromDir(std::shared_ptr<fs::LocalFileSystem>& local_fs,
+                            std::string dirpath, std::vector<fs::FileInfo>& rel_fpaths) {
+  // Define a selector for a recursive descent
+  fs::FileSelector selector;
+  selector.base_dir = dirpath;
+  selector.recursive = true;
+
+  ARROW_ASSIGN_OR_RAISE(auto file_infos, local_fs->GetFileInfo(selector));
+  std::move(file_infos.begin(), file_infos.end(), std::back_inserter(rel_fpaths));
```

Comment:
This will also give you subdirectories, is that what you want or should you filter them out?

Reply:
I think Weston must have misunderstood me then. I had code for omitting subdirectories, so I can just put that back in

Reply:
Ha, sorry :-)